src/server/components/dashboard/controllers.ts

```diff
@@ -317,20 +173,191 @@ export async function listsEditController(
 
     res.render("dashboard/lists-edit", {
       ...DEFAULT_VIEW_PROPS,
-      listCreated,
-      listUpdated,
+      publisher: { change: changeMsg },
       listId,
-      isPost,
-      error,
+      user: req.user?.userData,
       list,
       req,
       csrfToken: getCSRFToken(req),
     });
   } catch (error) {
+    const err = new HttpException(404, "404", "List could not be found.");
+    next(err);
+  }
+}
+
+export async function listsEditPostController(
+  req: Request,
+  res: Response,
+  next: NextFunction
+): Promise<void> {
+  try {
+    const removeButtonClicked = "userEmail" in req.body;
+
+    return removeButtonClicked
+      ? await listEditRemovePublisher(req, res)
+      : await listEditAddPublisher(req, res, next);
+
+  } catch (error) {
+    logger.error(`listsEditPostController error: ${(error as Error).message}`);
     next(error);
   }
 }
 
+export async function listEditAddPublisher(req: Request,
+  res: Response, next: NextFunction): Promise<void> {
+
+  const { listId } = req.params;
+  let error: Partial<QuestionError> = {};
+
+  const list: List | undefined = await findListById(listId);
+  const publisher: string = req.body.publisher;
+
+  const user = req.user;
+  const userIsSuperAdmin = user?.isSuperAdmin();
+
+  if (!userIsSuperAdmin || (listId === "new" && !userIsSuperAdmin))  {
+    const err = new HttpException(403, "403", "You are not authorized to access this list.");
+    return next(err);
+  }
+
+  if (!publisher || !isGovUKEmailAddress(publisher)) {
+    error = {
+      field: "publisher",
+      text:
+        !publisher
+          ? "You must indicated a publisher"
+          : "New users can only be example@fco.gov.uk, or example@fcdo.gov.uk",
+      href: "#publisher",
+    };
+  }
+
+  if (list?.jsonData.publishers.includes(publisher)) {
+    error = {
+      field: "publisher",
+      text: "This user already exists",
+      href: "#publisher",
+    };
+  }
+
+  const errorExists = ("field" in error);
+
+  if (errorExists) {
+    return res.render("dashboard/lists-edit", {
```

Comment:
@alisalaman-cyb, @RichardBray should I go through Post Redirect Get on a call?..

This is a separation of concerns principle. Post requests should just deal with "what a user wants to action". Rendering of a page (requesting a HTML resource!) should be a GET request. Redirecting with flash consolidates the two.